### PR TITLE
Several refactorations

### PR DESCRIPTION
**Pull request by fflorent**
## Problem

- The addition of new configurations was a bit tough;
- No test existed for upgrade from version 1.0.X
- The warning telling that the `garradin-ynh-deps` package does not exist is confusing, filter it out from stderr;
- installing dependencies using `ynh_add_fpm_config` is not advised anymore (see  https://github.com/YunoHost/yunohost/pull/1018)

## Solution

- Introduce:
  - `config.local.php` which requires the following files;
  - `config.local.yunohost.php` which is updated along with updates;
  - `config.local.user.php` which is preserved across updates so the user can edit it as they will;
- **Rational for the above choice**: the configuration file `config.local.php` previously stated that the user should not update the file, so we assume it is allowed to overwrite it.
- Update the check_process to check an update from the 1.0.6 version;
- Use `ynh_install_app_dependencies` as suggested now: https://github.com/YunoHost/yunohost/pull/1018

## PR Status

- [x] Code finished and ready to be reviewed/tested
- [x] The fix/enhancement were manually tested (if applicable)

## Automatic tests

Automatic tests can be triggered on https://ci-apps-dev.yunohost.org/ *after creating the PR*, by commenting "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!". (N.B. : for this to work you need to be a member of the Yunohost-Apps organization)

### CODE REVIEW

**Comment by fflorent** [comment]
!testme

File: conf/config.local.yunohost.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
+<?php
+
+/**
+ * Ce fichier représente un exemple des constantes de configuration
+ * disponibles pour Garradin.
```
Comment (fflorent) [inline conf/config.local.yunohost.php:5]: Not relevant.

**Comment by rodinux** [comment]
what about the warning ?
```
! General stuff, misc helper usage

    ! ynh_install_app_dependencies should also be in restore script 
```

**Comment by fflorent** [comment]
> what about the warning ?
> 
> ```
> ! General stuff, misc helper usage
> 
>     ! ynh_install_app_dependencies should also be in restore script 
> ```

That's a good point! I'll fix that today, thanks!

**Comment by rodinux** [comment]
I have try your code on a virtualbox yunohost, I think it nice ! Thanks for your job.

**Comment by rodinux** [comment]
There is something I can't explain. First I have try install garradin, the first installation in the configuration page  said me I have an 1.1.12 version, but it should have been an 1.1.14... I can't understand why ???

**Comment by fflorent** [comment]
> There is something I can't explain. First I have try install garradin, the first installation in the configuration page said me I have an 1.1.12 version, but it should have been an 1.1.14... I can't understand why ???

I get this:

**Comment by fflorent** [comment]
!testme

**Comment by rodinux** [comment]
> > There is something I can't explain. First I have try install garradin, the first installation in the configuration page said me I have an 1.1.12 version, but it should have been an 1.1.14... I can't understand why ???
> 
> I get this: 

Me too after upgrading with your code. I mean with the first install which may install the master branch, I get version 1.1.12 is installed, but it should be 1.1.14 ???

**Comment by rodinux** [comment]
Ok, we wait for the results, but I think I will push your PR soon, it makes sense.

**Pull request merged**